### PR TITLE
Button Component Testing

### LINKED ISSUE #488: Button Component Test
/components/Forms/Button

### PR DESCRIPTION
**Pull request by padam11**
code
button

### CODE REVIEW

File: cypress/component/Button.cy.jsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import Button from "@/components/Forms/Button";
+
+describe("Button.cy.jsx", () => {
+  it("text", () => {
+    const text = "Hello, world";
+    cy.mount(<Button text={text}>Hello, world</Button>);
```
Comment (shahdivyank) [inline cypress/component/Button.cy.jsx]: Button is a self contained component, doesn't need children

File: cypress/component/Button.cy.jsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    cy.get("button").should("contains.text", "Hello, world");
+  });
+
+  it("no text", () => {
+    cy.mount(<Button></Button>);
```
Comment (shahdivyank) [inline cypress/component/Button.cy.jsx]: self closing

File: cypress/component/Button.cy.jsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    cy.get("button").should("not.contain.text");
+  });
+
+  it("clickable", () => {
+    const onClick = cy.stub().as("clickHandler");
```
Comment (shahdivyank) [inline cypress/component/Button.cy.jsx]: dont need the alias

File: cypress/component/Button.cy.jsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    cy.mount(<Button onClick={onClick} />);
+    cy.get("button").click({ force: true });
```
Comment (shahdivyank) [inline cypress/component/Button.cy.jsx]: shouldnt be using force

File: cypress/component/Button.cy.jsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  });
+
+  it("no text", () => {
+    cy.mount(<Button></Button>);
+    cy.get("button").should("not.contain.text");
+  });
+
+  it("clickable", () => {
+    const onClick = cy.stub().as("clickHandler");
+    cy.mount(<Button onClick={onClick} />);
```
Comment (shahdivyank) [inline cypress/component/Button.cy.jsx]: add text

File: cypress/component/Button.cy.jsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+describe("Button.cy.jsx", () => {
+  it("text", () => {
+    const text = "Hello, world";
+    cy.mount(<Button text={text}>Hello, world</Button>);
+    cy.get("button").should("contains.text", "Hello, world");
+  });
+
+  it("no text", () => {
+    cy.mount(<Button></Button>);
+    cy.get("button").should("not.contain.text");
```
Comment (shahdivyank) [inline cypress/component/Button.cy.jsx]: use the proper method of getting the button

File: cypress/component/Button.cy.jsx [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    const text = "";
+    cy.mount(<Button text={text} />);
+    cy.get('[data-cy="-button"]').should("not.contain.text");
+  });
+
+  it("clickable", () => {
+    const text = "clickable";
+    const onClick = cy.stub();
+    cy.mount(<Button onClick={onClick} text={text} />);
+    cy.get('[data-cy="clickable-button"]').click();
```
Comment (shahdivyank) [inline cypress/component/Button.cy.jsx]: check if the button was clicked

**Review by menthy-wu — APPROVED** [review]

**Review by menthy-wu — APPROVED** [review]

**Review by shahdivyank — APPROVED** [review]

**Pull request merged**